spark-3.1-spanner-lib/src/test/java/com/google/cloud/spark/spanner/SpannerTestBase.java

```diff
@@ -121,6 +121,45 @@ private static void initDatabase() throws Exception {
 
               return null;
             });
+
+    // 3.2. Insert the Shakespeare data.
+    int maxValuesPerTxn = 35_000;
+    int MAX_BYTE_SIZE_PER_TXN = 1_048_576;
+
+    List<String> shakespearValues = TestData.shakespearValues;
+    for (int start = 0; start < shakespearValues.size(); ) {
+      int end = start + maxValuesPerTxn;
+      if (end > shakespearValues.size()) {
+        end = shakespearValues.size();
+      }
+
+      String sqlPrefix =
+          "INSERT INTO Shakespeare(id, word, word_count, corpus, corpus_date) VALUES";
+      int i = start;
+      int byteSize = sqlPrefix.length();
+      for (i = start; i < end && (i - start) < maxValuesPerTxn; i++) {
+        String currentValue = shakespearValues.get(i);
+        if (currentValue.length() + 1 + byteSize >= MAX_BYTE_SIZE_PER_TXN) {
+          break;
+        } else {
+          byteSize += currentValue.length() + 1;
+        }
+      }
+
+      end = i;
+      List<String> values = shakespearValues.subList(start, end);
+      start = end;
+
+      databaseClient
+          .readWriteTransaction()
```

Comment:
NIT: Maybe using write(partition)  is faster. But it's your choice.

Reply:
Currently that seems much more complex because I have to create Mutations and then insert values by their keys. The ReadWriteTransaction allows me to insert SQL statements as they are https://cloud.google.com/spanner/docs/modify-mutation-api and sure I could create mutations using Statement.of but this can suffice. Also currently javadoc.io https://javadoc.io/doc/com.google.cloud/google-cloud-spanner/latest/com/google/cloud/spanner/DatabaseClient.html is down 
<img width="1673" alt="Screenshot 2023-10-03 at 2 40 08 PM" src="https://github.com/GoogleCloudDataproc/spark-spanner-connector/assets/4898263/475fcacb-ac6a-49ba-9a12-4732860bad2c">